test/io/test_io.py

```diff
@@ -853,7 +853,7 @@ def test_metrics_include_schema_cache_fails(defaultenv, metapostgrest):
                 r'pgrst_schema_cache_loads_total{status="FAIL"} (\d+)', response.text
             ).group(1)
         )
-        assert metrics > 3.0
+        assert metrics == 1.0
```

Comment:
This test proves that this works. I didn't want to rely on the test mentioned in https://github.com/PostgREST/postgrest/issues/3523#issuecomment-2215053565 because it relies on a new feat and I want to backport this fix on v12.

Manually it can be confirmed that the above is fixed though.

```
$ PGRST_DB_SCHEMAS="x" postgrest-with-postgresql-15  -f test/spec/fixtures/load.sql postgrest-run

10/Jul/2024:20:56:36 -0500: Starting PostgREST 12.3 (pre-release)...
10/Jul/2024:20:56:36 -0500: Listening on port 3000
10/Jul/2024:20:56:36 -0500: Listening for notifications on the "pgrst" channel
10/Jul/2024:20:56:36 -0500: Successfully connected to PostgreSQL 15.7 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 13.2.0, 64-bit
10/Jul/2024:20:56:36 -0500: Failed to load the schema cache. {"code":"3F000","details":null,"hint":null,"message":"schema \"x\" does not exist"}

10/Jul/2024:20:56:37 -0500: Attempting to reconnect to the database in 1 seconds...
10/Jul/2024:20:56:37 -0500: Successfully connected to PostgreSQL 15.7 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 13.2.0, 64-bit
10/Jul/2024:20:56:37 -0500: Failed to load the schema cache. {"code":"3F000","details":null,"hint":null,"message":"schema \"x\" does not exist"}

10/Jul/2024:20:56:39 -0500: Attempting to reconnect to the database in 2 seconds...
10/Jul/2024:20:56:39 -0500: Successfully connected to PostgreSQL 15.7 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 13.2.0, 64-bit
10/Jul/2024:20:56:39 -0500: Failed to load the schema cache. {"code":"3F000","details":null,"hint":null,"message":"schema \"x\" does not exist"}
```